### ISSUE TITLE
Support stacking Rewriters from different compilation passes

### ISSUE THREAD

**Issue by alexeyr** [scalan/scalan#126]
We can explicitly reuse a Rewriter on each compilation step, but this is inconvenient. It would be useful to say that a given rewriter should be used on every following stage.

**Comment by aslesarenko** [comment]
Yes. Once we have first-class rewriters we can create a kind of DSL for
their configuration. But the rules should be clear, explicit and simple.
Otherwise it will be a mess.
And this configuration should be part of compiler passes configuration.
Everything in one place.
